Big sur update (#11)
dortania/dortania.github.io: _posts/2020-10-13-bigsur-new.md

We work quite closely with the community and developers to ensure information's correct, up-to-date and of the best standards. While not perfect in every way, we hope to be the go-to resource for reliable Hackintosh information.

And for the times our information is either outdated, missing context or generally needs improving, we have our bug tracker to allow the community to more easily bring attention to issues and speak directly with the authors:

* [Dortania's Bugtracker](https://github.com/dortania/bugtracker)

## Dortania's Build Repo

For those who either want to run the lastest builds of a kext or need an easy way to test old builds of something, Dortania's Build Repo is for you!

* [**Build Repo**](https://dortania.github.io/builds/)

Kexts here are built right after commit, and currently supports most of Acidanthera's kexts and some 3rd party devs as well. If you'd like to add support for more kexts, feel free to PR: [Build Repo source](https://github.com/dortania/build-repo)

## True legacy macOS Support!

> As of OpenCore's latest versioning, 0.6.2, you can now boot every version of x86-based builds of OS X/macOS! A huge achievement on GoldFish64's part, we now support every major version of kernel cache both 32 and 64-bit wise. This means machines like Yonah and newer should work great with OpenCore and you can even relive the old days of OS X like OS X 10.4!

Reviewer: ```suggestion
As of OpenCore's latest versioning, 0.6.2, you can now boot every version of x86-based builds of OS X/macOS! A huge achievement on [@Goldfish64](https://github.com/Goldfish64)'s part, we now support every major version of kernel cache both 32 and 64-bit wise. This means machines like Yonah and newer should work great with OpenCore and you can even relive the old days of OS X like OS X 10.4!
```

* Apple's use of "Apple silicon" hints that ARM is not actually what future macs will be running, instead we'll see highly customized chips *based* off ARM
  * For example, Apple will be heavily relying on hardware features such as [W^X](https://en.wikipedia.org/wiki/W%5EX), kernel memory protection, Pointer Auth, etc for security and thus both macOS and Applications will be dependant on it. This means hackintoshing on bare-metal(without a VM) will become extremely difficult without copious amounts of work
  * Also keep in mind Apple Silicon will no longer be UEFI-based like Intel Macs are, meaning a huge amount of work would also be required on this end as well

So while we may be heart broken the journey is coming to a stop in the somewhat near future, hackintoshing will still be a time piece in Apple's history. So enjoy it now while we still can, and we here at Dortania will still continuing supporting the community with our guides till the very end!

# Getting ready for macOS 11, Big Sur

This will be your short run down if you skipped the above:

* DRM is broken for many
  * Specifically for those relying on WhateverGreen for DRM

> * Many Ivy Bridge and Haswell SMBIOS were dropped

Reviewer 2: ```suggestion
* Some kexts are broken (like MacProMemoryNotificationDisabler) because Lilu user-patching has severely broke.
* Many Ivy Bridge and Haswell SMBIOS were dropped
```

* [New Kernel Requirements](#new-kernel-requirements)
* [Broken Kexts in Big Sur](#broken-kexts-in-big-sur)
* [MSI Navi installer Bug Resolved](#msi-navi-installer-bug-resolved)
* [New AMD OS X Kernel Patches](#new-amd-os-x-patches)
* [Other notable Hackintosh issues](#other-notable-hackintosh-issues)
  * [Several SMBIOS have been dropped](#several-smbios-have-been-dropped)
  * [Dropped hardware](#dropped-hardware)
  * [Extra long install process](#extra-long-install-process)
  * [X79 and X99 Boot issues](#x79-and-x99-boot-issues)
  * [New RTC requirements](#new-rtc-requirements)
  * [SATA Issues](#sata-issues)

## New Kernel Cache system: KernelCollections!

So for the past 15 years, macOS has been using the Prelinked Kernel as a form of Kernel and Kext caching. And with macOS Big Sur's new Read-only, snapshot based system volume, a new version of caching has be developed: KernelCollections!

How this differs to previous OSes:

* Kexts can no longer be hot-loaded, instead requiring a reboot to load with `kmutil`
* The Secure Boot and standard kernel are now one(ie. no more dedicated Immutable Kernel)
* Symbols are no longer required

> The last point is the most important part, as this is what we use for kext injection in OpenCore. Currently Apple has left symbols in place seemingly for debugging purposes however this is a bit worrying as Apple could outright remove symbols in later versions of macOS. But for macOS 11, Big Sur's cycle, we'll be good on that end however we'll be keeping an eye on macOS 12.

Reviewer 2: ```suggestion
The last point is the most important part, as this is what we use for kext injection in OpenCore. Currently Apple has left symbols in place seemingly for debugging purposes however this is a bit worrying as Apple could outright remove symbols in later versions of macOS. But for macOS 11, Big Sur's cycle, we'll be good on that end however we'll be keeping an eye on future releases of macOS.
```

Author: > Apple can drop support for symbols in macOS 11.x, not necessary 12

Very unlikely, as Apple rarely touches the their kernelcache system within an OS cycle and in all honesty wouldn't make sense especially if devs would be relying on them as well

## New Kernel Requirements

> With this update, the `AvoidRuntimeDefrag` Booter quirk in OpenCore broke. Because of this, the macOS kernel will fall flat when trying to boot. Reason for this is due to `cpu_count_enabled_logical_processors` requiring the MADT (APIC) table, and so OpenCore will now ensure this table is made accessible to the kernel. Users will however need a build of OpenCore 0.6.0 with commit [`bb12f5f`](https://github.com/acidanthera/OpenCorePkg/commit/9f59339e7eb8c213e84551df0fdbf9905cd98ca4) **or newer** to resolve this issue.

Reviewer 2: ```suggestion
With this update, the `AvoidRuntimeDefrag` Booter quirk in OpenCore broke. Because of this, the macOS kernel will fall flat when trying to boot. Reason for this is due to `cpu_count_enabled_logical_processors` requiring the MADT (APIC) table, and so OpenCore will now ensure this table is made accessible to the kernel. Users will however need a build of OpenCore 0.6.0 **or newer** to resolve this issue.
```

Author: Specific reason we shouldn't link a specific commit?

## Broken Kexts in Big Sur

Unfortunately with the aforementioned KernelCollections, some kexts have unfortunately broken or have been hindered in some way. The main kexts that currently have issues are anything relying on Lilu's userspace patching functionality:

* [WhateverGreen's](https://github.com/acidanthera/WhateverGreen) [DRM](https://dortania.github.io/OpenCore-Post-Install/universal/drm.html) and `-cdfon` patches
  * Most of WhateverGreen's functions are still working as they're kernelspace based
* [MacProMemoryNotificationDisabler](https://github.com/IOIIIO/MacProMemoryNotificationDisabler)
* [SystemProfilerMemoryFixup](https://github.com/Goldfish64/SystemProfilerMemoryFixup)

Thankfully most important kexts rely on kernelspace patcher which is now in fact working again.

## MSI Navi installer Bug Resolved

For those receiving boot failures in the installer due to having an MSI Navi GPU installed, macOS Big Sur has finally resolved this issue!

## New AMD OS X Kernel Patches

For those running on AMD-Based CPUs, you'll want to also update your kernel patches as well since patches have been rewritten for macOS Big Sur support:

> * [AMD OSX Kernel Patches](https://github.com/AMD-OSX/AMD_Vanilla)

Reviewer 2: ```suggestion
* [AMD OSX Kernel Patches](https://github.com/AMD-OSX/AMD_Vanilla/tree/experimental-opencore)
```

Author: This PR is to be merged with BigSur's release, and AMD OSX will be merging into mainline as well

Reviewer 2: However keep an eye on the patches

  * [Dortania's Build Repo](#dortania-s-build-repo)
  * [True legacy macOS Support!](#true-legacy-macos-support)
  * [Intel Wireless: More native than ever!](#intel-wireless-more-native-than-ever)

>   * [Clover's revival? A frankestien of a bootloader](#clovers-revival-a-frankestien-of-a-bootloader)

Reviewer: ```suggestion
  * [Clover's revival? A frankenstein of a bootloader](#clovers-revival-a-frankenstein-of-a-bootloader)
```

### X79 and X99 Boot issues

> With macOS 11, Big Sur, IOPCIFamily went through a decent rewriting causing many of X99 patches to break and many X79 boards fail to boot as well with IOPCIFamily kernel panics. Currently no solution is available so we recommend staying on macOS Catalina until issues have been resolved, however Acidanthera is actively looking into the situation and the guides will be updated with information when released.

Reviewer: ```suggestion
With macOS 11, Big Sur, IOPCIFamily went through a decent rewriting causing many of X99 patches to break and many X79 boards fail to boot as well with IOPCIFamily kernel panics. Currently no solution is available so we recommend staying on macOS Catalina until issues have been resolved, however Acidanthera is actively looking into the situation and the guides will be updated when new information is available.
```

* [Dortania's Build Repo](#dortania-s-build-repo)
* [True legacy macOS Support!](#true-legacy-macos-support)
* [Intel Wireless: More native than ever!](#intel-wireless-more-native-than-ever)

> * [Clover's revival? A frankestien of a bootloader](#clovers-revival-a-frankestien-of-a-bootloader)

Reviewer: ```suggestion
* [Clover's revival? A frankenstein of a bootloader](#clovers-revival-a-frankenstein-of-a-bootloader)
```

This will be your short run down if you skipped the above:

* Lilu's userspace patcher is broken

>   * Due to this many kext will break:

Reviewer: ```suggestion
  * Due to this many kexts will break:
```
